#1. 연습문제/과제풀이

# A: 6장 신규노선 항공운임예측 풀기

먼저, 데이터 파일을 불러옴.

In [2]:
from google. colab import files
uploaded=files.upload()

Saving Airfares.csv to Airfares.csv


# a. 수치형 예측 변수들과 결과 변수인 운임(FARE)에 대한 표본 만들고 운임과 예측 변수들 사이의 상관도를 조사하시오. 어떤 예측 변수와 운임을 예측하는 데 최적의 예측 변수인가?

1. 수치형 변수 선택

수치형(연속형) 변수만 대상으로 해야 하므로 범주형 변수는 제외하고 다음과 같은 변수들을 선택함:

COUPON, NEW, HI, S_INCOME, E_INCOME, S_POP, E_POP, DISTANCE, PAX

2. 운임(FARE)과의 상관계수 계산

상관계수(Pearson correlation coefficient)를 FARE에 대해 계산하면, 어떤 변수가 운임과 직선적으로 얼마나 강한 관계를 가지는지 알 수 있음.

In [3]:
import pandas as pd

df = pd.read_csv("Airfares.csv")

# 수치형 변수만 필터링
numeric_vars = ['COUPON', 'NEW', 'HI', 'S_INCOME', 'E_INCOME', 'S_POP', 'E_POP', 'DISTANCE', 'PAX', 'FARE']
corr_matrix = df[numeric_vars].corr()

# 운임(FARE)과 다른 변수들 간의 상관계수 정렬
fare_corr = corr_matrix['FARE'].drop('FARE').sort_values(ascending=False)
print(fare_corr)

DISTANCE    0.670016
COUPON      0.496537
E_INCOME    0.326092
E_POP       0.285043
S_INCOME    0.209135
S_POP       0.145097
NEW         0.091730
HI          0.025195
PAX        -0.090705
Name: FARE, dtype: float64


운임(FARE)을 예측하는 데 있어 가장 효과적인 수치형 변수들을 파악하기 위해, 각 변수와 FARE 간의 Pearson 상관계수를 계산하였다. 상관계수가 클수록 두 변수 간의 선형적 관계가 강하다고 해석할 수 있다.

* DISTANCE(거리) 변수는 운임과 가장 높은 양의 상관계수(0.6700)를 보여주며,
운임 예측에 가장 효과적인 변수로 나타났다. 이는 항공 노선의 거리가 멀수록 평균 항공료가 높아지는 일반적인 경제 원칙과도 부합한다.

* COUPON, E_INCOME도 다소 의미 있는 양의 상관을 보이며, 운임에 일정 수준의 영향을 주는 변수로 판단된다.

* 반면 PAX(탑승객 수)는 음의 상관(-0.0907)을 보이며, 탑승객 수가 많을수록 운임이 낮아지는 경향을 나타낸다.

* HI(시장 집중도)와 NEW(신규 항공사 수) 등은 상관도가 낮아 단일 변수로서의 예측력은 크지 않을 것으로 보인다.

결론:

운임(FARE)을 예측하는 데 **가장 효과적인 수치형 변수는 DISTANCE**이며, 그 다음으로 COUPON과 E_INCOME이 유의미한 예측력을 보였다.

# b. 범주형 예측 변수들(예: SW 등)에 대해 범주별로 평균 운임의 차이와 분산의 패턴을 분석하고, 어떤 범주형 예측 변수가 운임 예측에 가장 좋은지를 평가하라.

범주형 변수는 주로 다음과 같음:

* SW : Southwest 항공 운항 여부 (Yes/No)

* VACATION : 휴양지 여부 (Yes/No)

* SLOT, GATE : 공항의 슬롯/게이트 제한 여부 (Free/Controlled)

* NEW도 범주형으로 처리 가능

이 중에서 **SW, VACATION, SLOT, GATE**를 주요 분석 대상으로 삼음.

그 다음, 각 범주형 변수의 그룹별 평균 운임 확인

In [4]:
# SW 변수에 따른 평균 운임 확인
print(df.groupby('SW')['FARE'].mean())
print(df.groupby('SW')['FARE'].std())

print(df.groupby('VACATION')['FARE'].mean())
print(df.groupby('VACATION')['FARE'].std())

print(df.groupby('SLOT')['FARE'].mean())
print(df.groupby('GATE')['FARE'].mean())

SW
No     188.182793
Yes     98.382268
Name: FARE, dtype: float64
SW
No     70.558360
Yes    44.795629
Name: FARE, dtype: float64
VACATION
No     173.552500
Yes    125.980882
Name: FARE, dtype: float64
VACATION
No     81.086891
Yes    44.102634
Name: FARE, dtype: float64
SLOT
Controlled    186.059396
Free          150.825680
Name: FARE, dtype: float64
GATE
Constrained    193.129032
Free           153.095953
Name: FARE, dtype: float64


따라서 결론은 이러함.

* SW 변수는 Yes일 때 평균 운임이 훨씬 낮아짐 → 항공사 경쟁 효과로 해석 가능. 운임 예측에 매우 유의미.

* **SLOT, GATE**는 제약이 있는 경우 운임이 높음 → 공항 인프라의 제약이 가격에 영향을 미친다는 해석.

* **VACATION**은 유의미한 차이를 보일 수도 있지만 경우에 따라 차이가 작을 수 있음.

결론적으로 운임 예측에 **가장 유효한 범주형 변수는 SW**로 나타났으며, SLOT, GATE도 유의미한 설명력을 가진다.